caffe2/intra_op_parallel/parallel_benchmark_utils.py

```diff
@@ -0,0 +1,218 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import copy
+
+from caffe2.proto import caffe2_pb2
+from caffe2.python import core
+
+
+def _get_replicated_blob_name(blob_name, numa_node_id):
+    return (
+        blob_name
+        if numa_node_id == 0
+        else "replica:{}/{}".format(numa_node_id, blob_name)
+    )
+
+
+def _get_sharded_blob_name(blob_name, numa_node_id):
+    return "shard:{}/{}".format(numa_node_id, blob_name)
+
+
+def _create_copies_for_replica(blob_name, num_numa_nodes):
+    return [
+        core.CreateOperator(
+            "Copy",
+            [blob_name],
+            _get_replicated_blob_name(blob_name, i),
+            device_option=caffe2_pb2.DeviceOption(numa_node_id=i),
+        )
+        for i in range(1, num_numa_nodes)
+    ]
+
+
+def _create_prepand_dim_split_merge_dim(blob_name, num_numa_nodes):
+    default_device_option = caffe2_pb2.DeviceOption(numa_node_id=0)
+    new_ops = [
+        core.CreateOperator(
+            "PrependDim",
+            [blob_name],
+            blob_name,
+            dim_size=num_numa_nodes,
+            device_option=default_device_option,
+        ),
+        core.CreateOperator(
+            "Split",
+            [blob_name],
+            [
+                _get_sharded_blob_name(blob_name, numa_node_id)
+                for numa_node_id in range(num_numa_nodes)
+            ],
+            axis=0,
+            device_option=default_device_option,
+        ),
+        core.CreateOperator(
+            "MergeDim", [blob_name], blob_name, device_option=default_device_option
+        ),
+    ]
+    new_ops += [
+        core.CreateOperator(
+            "MergeDim",
+            [_get_sharded_blob_name(blob_name, numa_node_id)],
+            _get_sharded_blob_name(blob_name, numa_node_id),
+            device_option=default_device_option,
```

Comment:
I just wonder why "MergeDim" for all numa-nodes use "device_option=default_device_option" and not use "device_option=caffe2_pb2.DeviceOption(numa_node_id=numa_node_id)" like "FC" ops ? 
Thanks